### ISSUE TITLE
Containerization Followup: htpasswd on atomic host when using htpasswd auth provider

### ISSUE THREAD

**Issue by sdodson** [openshift/openshift-ansible#1071]
Followup for #808 
In RHEL installs we install http-utils for users so that when they're using htpasswd auth it's easy to generate password files. It'd be cool if we were able to do this on atomic host as well perhaps using rhscl/httpd-24-rhel7

**Comment by ibotty** [comment]
Isn't that image a little heavy?

**Comment by sdodson** [comment]
It was the only httpd image on registry.access.redhat.com when I looked. Did you have another image in mind?

I guess as an alternative to pulling an apache httpd image we could find a python implementation of htpasswd and copy that to /usr/local/bin/.

**Comment by ibotty** [comment]
> It was the only httpd image on registry.access.redhat.com when I looked. Did you have another image in mind?

I did not have one in mind. I would have rolled a minimal one or
let users use rhel-tools (or fedora-tools) and install it there.

> I guess as an alternative to pulling an apache httpd image we could find a python implementation of htpasswd and copy that to /usr/local/bin/. 

Most seem to use passlib, which is not installed on atomic hosts afaict.

**Comment by cgwalters** [comment]
Doesn't need to live on the host system, FWIW I just did:

```
atomic run rhel7/rhel-tools
yum -y install /usr/bin/htpasswd
htpasswd -c /host/etc/origin/master/htpasswd
```

Note the use of `/host`.

**Comment by sdodson** [comment]
hmm, a generic container that's got an entrypoint that takes commands and
arguments, uses yum to install them on the fly and then run the provided
command...

**Comment by detiber** [comment]
@sdodson for one-shot tasks like the htpasswd file I agree...  For something like the oc/oadm client I think we'd want to stick with the client container.

**Comment by vrutkovs** [comment]
`htpasswd` is now being run locally, fixed in #3922
